### PR TITLE
Allow two-digit-max-year customization in patterns

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

**Comment by jskeet** [comment]
It's been a while, but I think this is ready now :) Lots of copy/paste, so if you find any typos in docs etc, I'll no doubt need to fix them everywhere...

File: src/NodaTime/Text/InstantPattern.cs [hunk 1]
```diff
@@ -69,16 +69,22 @@ private static class Patterns
         /// <value>The value used as a template for parsing.</value>
         public Instant TemplateValue { get; }
 
+        /// <summary>
+        /// Maximum two-digit-year in the template to treat as the current century.
+        /// </summary>
+        public int TwoDigitYearMax { get; }
```
Comment (malcolmr) [inline src/NodaTime/Text/InstantPattern.cs:75]: optional: worth adding a `<value>` per above?

File: src/NodaTime/Text/InstantPattern.cs [hunk 2]
```diff
@@ -189,6 +196,14 @@ private static InstantPattern Create(string patternText, NodaFormatInfo formatIn
         /// <param name="newTemplateValue">The template value for the new pattern, used to fill in unspecified fields.</param>
         /// <returns>A new pattern with the given template value.</returns>
         public InstantPattern WithTemplateValue(Instant newTemplateValue) =>
-            Create(PatternText, FormatInfo, newTemplateValue);
+            Create(PatternText, FormatInfo, newTemplateValue, TwoDigitYearMax);
+
+        /// <summary>
+        /// Creates a pattern like this one, but with a different <see cref="TwoDigitYearMax"/> value.
+        /// </summary>
+        /// <param name="twoDigitYearMax">The value to use for <see cref="TwoDigitYearMax"/> in the new pattern.</param>
```
Comment (malcolmr) [inline src/NodaTime/Text/InstantPattern.cs:204]: worth adding something like "The value (0-99) to use..." to make the range clear?

Separately, where do we document the default values for this type? Probably we should document the default value for this new property?

**Review by malcolmr — APPROVED** [review]
LGTM

File: src/NodaTime/Text/InstantPattern.cs (hunk 1 above)
Comment (jskeet) [reply src/NodaTime/Text/InstantPattern.cs:75]: Done.

File: src/NodaTime/Text/InstantPattern.cs (hunk 2 above)
Comment (jskeet) [reply src/NodaTime/Text/InstantPattern.cs:204]: Done. Will add the default into the description of the property.

**Comment by jskeet** [comment]
Thanks for the review - will merge when I've finished tweaking the docs, and when it's gone greeen.

**Pull request merged**